src/ddprof_process.cc

```diff
@@ -43,10 +42,10 @@ DDRes Process::read_cgroup_ns(pid_t pid, std::string_view path_to_proc,
                               CGroupId_t &cgroup) {
   cgroup = Process::kCGroupNsError;
   std::string const path =
-      string_format("%s/proc/%d/ns/cgroup", path_to_proc.data(), pid);
-  char buf[k_max_buf_cgroup_link];
-  ssize_t const len = readlink(path.c_str(), buf, k_max_buf_cgroup_link - 1);
-  if (len == -1) {
+      absl::Substitute("$0/proc/$1/ns/cgroup", path_to_proc, pid);
```

Comment:
absl cat is usually the preferred API ?
From documentation
```
Note however, that absl::Substitute(), because it requires parsing a format string at run-time, is slower than absl::StrCat(). Choose Substitute() over StrCat() only when code clarity is more important than speed.
```

Reply:
If you want maximum performance, yes, `StrCat` should be preferred to `Substitute`, but it is a lot less readable.
The decision tree is something like:
- do I need formatting ?
  - yes: use `absl::StrFormat`
  - no: do I want maximum performance (at the expense of readability) ?
     - yes: use `absl:StrCat`
     - no: use `absl::Substitute`

Reply:
Replaced this occurence with `absl::StrCat`